backend/test/classes.controller.e2e-spec.ts

```diff
@@ -0,0 +1,112 @@
+import { INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { getApp } from "./helper";
+import { classes, defaultAdmin, defaultTeacher, users } from "test/seed";
+
+const checkClassesInList = (expectedClasses, returnedClasses): void => {
+  expect(returnedClasses).toHaveLength(expectedClasses.length);
+  returnedClasses.forEach((returnedClass) => {
+    const { teacherId, ...referenceClass } = expectedClasses.find(
+      (c) => c.id == returnedClass.id,
+    );
+    expect(returnedClass).toMatchObject(referenceClass);
+
+    const {
+      type: _ignore_type,
+      email: _ignore_email,
+      ...referenceTeacher
+    } = users.find((u) => u.id === teacherId)!;
+    expect(returnedClass.teacher).toMatchObject(referenceTeacher);
+  });
+};
```

Comment:
This can probably be done better, I would like to develop a library of matchers, but I believe it should be "good enough" for now :-)